src/cwe_checker_lib/src/ghidra_pcode/pcode_operations.rs

```diff
@@ -94,3 +94,56 @@ impl ExpressionType {
         }
     }
 }
+
+impl JmpType {
+    pub fn into_ir_branch(&self, target: Tid) -> Jmp {
+        if matches!(self, JmpType::BRANCH) {
+            Jmp::Branch(target)
+        } else {
+            panic!("Not a branch operation")
+        }
+    }
+
+    pub fn into_ir_cbranch(&self, target: Tid, condition: Expression) -> Jmp {
+        if matches!(self, JmpType::CBRANCH) {
+            Jmp::CBranch { target, condition }
+        } else {
+            panic!("Not a conditional branch operation")
+        }
+    }
+
+    pub fn into_ir_return(&self, expression: Expression) -> Jmp {
+        if matches!(self, JmpType::RETURN) {
+            Jmp::Return(expression)
+        } else {
+            panic!("Not a return operation")
+        }
+    }
+
+    pub fn into_ir_branch_indirect(&self, target: Expression, return_: Option<Tid>) -> Jmp {
+        if matches!(self, JmpType::CALLIND) {
+            Jmp::CallInd { target, return_ }
+        } else {
+            panic!("Not a call indirect operation")
+        }
+    }
+
+    pub fn into_ir_call(&self, target: Tid, return_: Option<Tid>) -> Jmp {
+        if matches!(self, JmpType::CALL) {
+            Jmp::Call { target, return_ }
+        } else {
+            panic!("Not a call operation")
+        }
+    }
+
+    pub fn into_ir_call_other(&self, description: String, return_: Option<Tid>) -> Jmp {
+        if matches!(self, JmpType::CALL) {
+            Jmp::CallOther {
+                description,
+                return_,
+            }
+        } else {
+            panic!("Not a call operation")
+        }
+    }
+}
```

Comment:
Is this a typo that you test for `JmpType::CALL` instead of `CALLOTHER`?

Reply:
Good catch!